language/diem-framework/modules/0L/TowerState.move

```diff
@@ -503,11 +481,23 @@ module TowerState {
     // returns the number of proofs for a miner in the current epoch
     public fun get_count_in_epoch(miner_addr: address): u64 acquires TowerProofHistory {
```

Comment:
I feel like you need to store the epoch that get_count_in_epoch was last called in and potentially count over multiple epochs.

You might be quietly allowing state to corrupt if a miner misses an epoch

Reply:
Every time there is an attempt to send a miner tx, we lazily reset the the epoch counter.

https://github.com/OLSF/libra/blob/ratelimit-issue/language/diem-framework/modules/0L/TowerState.move#L250-L256

And it uses the latest_epoch_mining of the user to check if we need a reset:

https://github.com/OLSF/libra/blob/ratelimit-issue/language/diem-framework/modules/0L/TowerState.move#L494-L499